pkg/endpoint/bpf.go

```diff
@@ -180,7 +180,9 @@ func (e *Endpoint) writeHeaderfile(prefix string, owner Owner) error {
 	fw.WriteString(" */\n\n")
 
 	// If there hasn't been a policy calculated yet, we need to be sure we drop all packets, but only if policy enforcement is enabled for the endpoint / daemon.
-	if !e.PolicyCalculated && e.Opts.IsEnabled(OptionPolicy) && owner.PolicyEnforcement() != NeverEnforce {
+	if !e.PolicyCalculated &&
+		(e.Opts.IsEnabled(OptionIngressPolicy) || e.Opts.IsEnabled(OptionEgressPolicy)) &&
```

Comment:
Hi Thomas changing this to 
  if !e.PolicyCalculated && owner.PolicyEnforcement() != NeverEnforce {
            fw.WriteString("#define DROP_ALL\n")
     } as you suggested, gives a cilium - consul timeout (>30s timeout error). I have messaged you offline with the logs. I am reverting this back to original logic and debugging it right now. Any suggestions why this could be the case? I am not sure if that's the reason why the existing logic was that way.

Reply:
@tgraf  I reconfirmed by running tests locally too, that this check is indeed causing the timeout. I will continue debugging it, but if this feature is blocking the release, I can follow this up in a separate PR/GH issue. Please let me know!